### ISSUE TITLE
Unresolvable 'new' Indentation Rule conflict introduced in 8.38

### ISSUE THREAD

**Issue by jcamerin** [checkstyle/checkstyle#9071]
The following code snippet exists in my codebase and passes Checkstyle using 8.37 rules.  When run against the 8.38 rules, I get violations due to updated indentation rules in the release: "'operator new' has incorrect indentation level 24, expected level should be one of the following: 28, 32."

```
                redirectAttributes.addFlashAttribute("globalMessage",
                        new GenericResponse(ResponseStatus.ERROR, messages.getMessage("data.trigger.enable.failure",
                        new Object[] { ERROR_MSG })));
```

I changed the indentation level to 28, and got this Checkstyle rule violation:
"'new' has incorrect indentation level 28, expected level should be 24."
```
                redirectAttributes.addFlashAttribute("globalMessage",
                        new GenericResponse(ResponseStatus.ERROR, messages.getMessage("data.trigger.enable.failure",
                            new Object[] { ERROR_MSG })));
```

I also tried changing the indentation level to 32, and got this Checkstyle rule violation:
"'new' has incorrect indentation level 32, expected level should be 24."
```
                redirectAttributes.addFlashAttribute("globalMessage",
                        new GenericResponse(ResponseStatus.ERROR, messages.getMessage("data.trigger.enable.failure",
                                new Object[] { ERROR_MSG })));
```

[checkstyle.xml.zip](https://github.com/checkstyle/checkstyle/files/5622635/checkstyle.xml.zip)

**Comment by Abhishek-kumar09** [comment]
Thanks @jcamerin  for raising this issue, could you provide the full function or class body to investigate further.

**Comment by jcamerin** [comment]
Full function
```
    public String enable(@PathVariable final Integer triggerId, final RedirectAttributes redirectAttributes) {
        final EventTrigger trigger = triggerService.findTrigger(triggerId);

        if (trigger.getInitialAction() != null) {
            try {
                triggerService.enable(trigger);
                redirectAttributes.addFlashAttribute("globalMessage", new GenericResponse(ResponseStatus.SUCCESS,
                        messages.getMessage("data.trigger.enable.success")));
            } catch (final MessageGearsServiceException e) {
                logger.warn("Attempt to subscribe to %s failed: " + e.getMessage(), trigger.getActivityType());
                redirectAttributes.addFlashAttribute("globalMessage",
                        new GenericResponse(ResponseStatus.ERROR, messages.getMessage("data.trigger.enable.failure",
                                new Object[] { ERROR_MSG })));
            }
        } else {
            redirectAttributes.addFlashAttribute("globalMessage", new GenericResponse(ResponseStatus.ERROR,
                    messages.getMessage("data.trigger.enable.failure", new Object[] { "No actions associated" })));
        }

        return BASE_REDIRECT + "admin/event/event_trigger/" + triggerId;
    }
```

**Comment by Abhishek-kumar09** [comment]
I guess you are checking agains some different config, as I am checking agains default indentation profile and everything looks fine there, with no error at all.
<details>
  <summary>Your code</summary>

```java
class T {
    public String enable(@PathVariable final Integer triggerId, final RedirectAttributes redirectAttributes) {
        final EventTrigger trigger = triggerService.findTrigger(triggerId);

        if (trigger.getInitialAction() != null) {
            try {
                triggerService.enable(trigger);
                redirectAttributes.addFlashAttribute("globalMessage", new GenericResponse(ResponseStatus.SUCCESS,
                        messages.getMessage("data.trigger.enable.success")));
            } catch (final MessageGearsServiceException e) {
                logger.warn("Attempt to subscribe to %s failed: " + e.getMessage(), trigger.getActivityType());
                redirectAttributes.addFlashAttribute("globalMessage",
                        new GenericResponse(ResponseStatus.ERROR, messages.getMessage("data.trigger.enable.failure",
                            new Object[] { ERROR_MSG })));
            }
        } else {
            redirectAttributes.addFlashAttribute("globalMessage", new GenericResponse(ResponseStatus.ERROR,
                    messages.getMessage("data.trigger.enable.failure", new Object[] { "No actions associated" })));
        }

        return BASE_REDIRECT + "admin/event/event_trigger/" + triggerId;
    }
}
```
</details>

config: 
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
    <module name="Indentation">
    </module>
    </module>
</module>

```

output: 
```
java -jar checkstyle-8.38-SNAPSHOT-all.jar -c config.xml t5.java 
Starting audit...
Audit done.
```

It will be helpful to us if you could share your config too.

**Comment by jcamerin** [comment]
Ah, good point...
Attached.

[checkstyle.xml.zip](https://github.com/checkstyle/checkstyle/files/5622635/checkstyle.xml.zip)

**Comment by Abhishek-kumar09** [comment]
Thanks I will see and get back to you with fix.

**Comment by Abhishek-kumar09** [comment]
Test.java
```
class T {
    public String test() {
        redirectAttributes.addFlashAttribute("globalMessage",
                new GenericResponse(ResponseStatus.ERROR, messages.getMessage("data.trigger.enable.failure",
                        new Object[] { ERROR_MSG })));
    }
}
```

config:
```
<module name="Indentation">
      <property name="basicOffset" value="4"/>
      <property name="braceAdjustment" value="0"/>
      <property name="caseIndent" value="4"/>
      <property name="throwsIndent" value="4"/>
      <property name="lineWrappingIndentation" value="8"/>
      <property name="forceStrictCondition" value="true" />
</module>
```

# checkstyle 8.38: 
```
Starting audit...
[ERROR] /home/abhishhek/playground/Test.java:5:25: 'new' has incorrect indentation level 24, expected level should be 16. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```
when indented on 16, it generates new error:
```
Starting audit...
[ERROR] /home/abhishhek/playground/Test.java:5:17: 'operator new' has incorrect indentation level 16, expected level should be one of the following: 20, 24. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```
# Checkstyle 8.37 and below:
```
Starting audit...
[ERROR] /home/abhishhek/playground/Test.java:5:25: 'new' has incorrect indentation level 24, expected level should be 16. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```
when indented on 16, everything passes: (which is desired).

**Comment by romani** [comment]
@Abhishek-kumar09, please follow strict on issue template

**Comment by Abhishek-kumar09** [comment]
> @Abhishek-kumar09, please follow strict on issue template

I didn't do `javac Test.java`, since I wanted to have the original code from the creater, and compling it requires lots of more code.
It is syntaxically correct, and thus no error thrown while processing the file with checkstyle.

**Comment by romani** [comment]
@Abhishek-kumar09 ,

> It is syntaxically correct,

after numerous case where we approved issue on invalid compilation, we tried to not rely on humans ability to do this.
Checkstyle can parse incorrect files and generate bad AST, so Check behavior might be weird.
in addition it is very helpful demand as you can copy input to tests quickly.
All code can changed to avoid extra types and variables in input and focus on what is matter by code structure based on jdk classes.

**Comment by Abhishek-kumar09** [comment]
@romani Done, please have  look.

**Comment by romani** [comment]
Fix is merged

**Comment by tmortagne** [comment]
This seems to have caused an unexpected side effect in the default behavior of IndentationCheck.

The following line used to be valid with Checkstyle 3.37: https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-13.0/xwiki-platform-core/xwiki-platform-icon/xwiki-platform-icon-default/src/main/java/org/xwiki/icon/internal/DefaultIconSetManager.java#L129

and is now failing with

> 'new' has incorrect indentation level 16, expected level should be 20.

Is this expected (a bit annoying since it's not matching the behavior of Eclipse's formatter anymore), or should I create a new issue as a regression ?

My configuration is https://github.com/xwiki/xwiki-commons/blob/xwiki-commons-13.0/xwiki-commons-tools/xwiki-commons-tool-verification-resources/src/main/resources/checkstyle.xml#L205

**Comment by Abhishek-kumar09** [comment]
@tmortagne Really sorry for the problem you faced after the merge, Indentation is always tricky and some corners are left even after extensive testings. If you can please create  a new issue so that we can work on your issue as soon as possible.

**Comment by tmortagne** [comment]
Sure, thanks for the confirmation @Abhishek-kumar09
